### ISSUE TITLE
Trying to emulate a larger SSD size

### ISSUE THREAD

**Issue by liorgordon** [ucare-uchicago/femu#19]
Hey there,
I'm currently trying to use your system with an SSD size of 16GB while running on the black-box mode.
I've tried changing it inside the black-box script and inside the ssd1.conf file but I keep getting the error message "exceed sector number".
what changes should i make in the ssd1.conf file so it'll be compatible to the size I've set in the NVMEIMGSZ variable inside the black-box script ?

**Comment by huaicheng** [comment]
Hi,

If your disk image size matches the size in the configuration file, it should be fine.
For the configuration file, the total simulated SSD size can be calculated as 
sim_ssd_sz = # of flash-chips * # of blocks per chip * # of pages per block * # of sectors per page * sector size.  (assuming # of planes per chip is 1). 

Make sure sim_ssd_sz * (1 - OVP) >= 16GB here. 

The blackbox FTL you are using is no longer maintained, we created a new FTL specifically designed for FEMU model, checkout the newest commit in master branch. I suggest you use the new one. We will add some documents later.

**Comment by liorgordon** [comment]
thank you for your quick response.
I decided to multilply the number of blocks per chip by 16 in order to get the size I wanted and  checkedout to the lastest commit like you suggested but I still wasn't able to create the larger ssd.

since I've clearly did something wrong i thought it would be best if I'd eleborate on what I've done.
I've created a vssd1.raw file with a size of 16GB using the qemu-img create command and put it in the images dir. I've also changed the BLOCK_NB var in the vssd1.conf file from 16 to 256.
when running the black box script and using the fdisk -l command (from within the femu vm) I could see the sda1 device and the nvme0n1 device but it's size was still 1 GB.
obivously I've misunderstood something in this process and I'd appriciate it if you'll tell me what am I missing here.

also, I didn't quite understand what you meant here:
Make sure sim_ssd_sz * (1 - OVP) >= 16GB here.

thank you so much for your help.

**Comment by huaicheng** [comment]
> Make sure sim_ssd_sz * (1 - OVP) >= 16GB here.

Oh, sorry, you can ignore this part ..

I'm gonna close this issue, just in case u have more questions, feel free to re-open it, thx!

**Comment by theanoli** [comment]
What is the relationship between the device size as given in the QEMU command and and the device size as hard-coded into ftl.c? If the passed-in device size is smaller than the hard-coded device size, then is the difference equivalent to overprovisioned device capacity? 

A related issue: I set them to the same thing (devsz_mb=2048, which the FEMU OS reports as 2147483648 bytes, and the appropriate settings in ftl.c) and I see the following: `RD-ERRRRRRRRRR,start_lpn=524287,end_lpn=524288,tt_pgs=524288`. (I also checked the read size and it's 8 sectors.) What is going wrong here---shouldn't the ending LPN wrap around?
